docs/product-repo-plan-proposal.md

```diff
@@ -0,0 +1,56 @@
+# Enabling Performance Testing in the Product Repos
+
+## Value Statement
+
+We want performance testing to be consistent across all product repos that we test.
+All repos (in particular, coreclr and corefx) should run in the same way with the same benchmarks so that we can easily determine where a regression occured.
+
+## Promises
+
+- A consistent AzDO yaml template in arcade for performance testing that can be used across the product repositories with minimal set up for product repo owners.
+- A helix proj file that runs all of the benchmarks and splits them in a sane way
+
+## Success Metric(s)
+
+- Product repositories use the provided yaml template to run all performance testing
+- Product repositories do not try to skirt the yaml template to run performance testing only on the tests they care about
+- Setup time for adding performance testing is as simple as adding a new job that references the yaml template that we provide
+
+## Procedural
+
+- Combine the yaml we already use and, likely, the yaml that has been worked on in coreclr (why reinvent the wheel), with any additional logic necessary for pulling down and running the performance repository in product repos
+  - yaml will need to include any additional parameters/knobs that the product repos may want to turn (ie tiered/non-tiered/minopts, pgo/no-pgo, etc)
+  - yaml will need to include all uploading to benchview logic
+  - yaml should be as barebones as possbile with most of the work done in the proj file
+  - yaml should allow sending of just the performance repo as well as sending the performance repo and the core_root directory (for repos that need to specify that)
```

Comment:
How does this square with our conversation around reworking things so that we can send a sane correlation package for all runs/scenarios/etc?

Reply:
My thought is that we would want to be able to share the yaml between the feature repos AND the performance repo, though they will have to send things slightly differently. In the performance repo, we would send just the benchmarks/sane correlation package for the benchmarks that are going to be run as the work item directory. In the feature repos, they will send the sane correlation package as the work item directory, and core_root as the correlation directory. They will always have to send more than we will.